### PR TITLE
added initial design document for an OMEGA Time Manager

### PR DESCRIPTION
**Pull request by philipwjones**
Added the initial document for a Time Manager, managing the simulation time and related alarms, calendars, etc. for the OMEGA model.

### CODE REVIEW

File: components/omega/doc/design/TimeMgr.md [hunk 1]
```diff
@@ -0,0 +1,970 @@
...
+
+## 4 Design
+
+The time manager design will follow an ESMF-like approach,
+though simplified and not requiring linking to the full
+ESMF library. Previous simplifications were performed for
+WRF by John Michalakes in Fortran and are used in MPAS, E3SM.
+A similar simplification was performed by Phil Jones from the
+ESMF C++ version for the SciDAC CANGA project and will be used
+here.
```
Comment (mark-petersen) [inline components/omega/doc/design/TimeMgr.md:102]: @philipwjones just to confirm, this means that the work proposed here for implementing the time manager is in the interface and creating the classes, but that the internal calendar math will be provided by a subset of the ESMF libraries? That sounds great.

Comment (philipwjones) [reply components/omega/doc/design/TimeMgr.md:102]: @mark-petersen  Actually, no. There isn't a way to do that. But I had already done a simplified C++ version for CANGA that I will be using here.

File: components/omega/doc/design/TimeMgr.md [hunk 2]
```diff
@@ -0,0 +1,970 @@
...
+simulations, the model must support years that extend beyond the
+4-digit representation and support negative years to represent
+years Before the Common Era (BCE)
+
+### 2.6 Required: Time string representation
+
+For model output or metadata, the model must be able to
+represent a time in accepted formats. Following ISO standards
+(eg ISO 8601 and prior), these will typically represent time
+in decreasing intervals, eg `YYYYYY-MM-DD HH:MM:SS.SSSSS` though
```
[the remark this answers is not in the document]
Comment (philipwjones) [reply components/omega/doc/design/TimeMgr.md:73]: Yes, I have preserved the fractional seconds.

File: components/omega/doc/design/TimeMgr.md (hunk 1 above)
Comment (mark-petersen) [reply components/omega/doc/design/TimeMgr.md:102]: Great! The important thing is the underlying library work is already done. Thanks!

**Review by mark-petersen — APPROVED** [review]
Looks good, thanks!

**Comment by vanroekel** [comment]
@philipwjones thanks for the design doc.  It makes a lot of sense to me and was very clear.  One quick question -- are you going to add anything to the testing portion?  Seems pretty straightforward to test a time manager, but am curious about your plans for section 5.

**Comment by philipwjones** [comment]
@vanroekel  Doh! Looks like I forgot to put that in. I do have an extensive unit test for the aforementioned CANGA version and I think I meant to look at it to add to the testing section and I will do so now.

**Comment by philipwjones** [comment]
@vanroekel I added a testing section - thanks for catching.

**Review by vanroekel — APPROVED** [review]
This looks good.  I especially enjoyed the day recap from 4173 BC

Thanks @philipwjones!

**Pull request merged**